### PR TITLE
State Validation Fix

### PR DESCRIPTION
**Pull request by rithiksharma100**
Bug: Getting state is not supported error on the iframe for the user login flow as checkout gets created without state.
Fix: Removed state to state-code conversion if it is empty

### CODE REVIEW

File: includes/simpl_integration/class-cart-response.php [hunk 1]
```diff
@@ -63,11 +63,13 @@ protected function convert_address_response($address) {
         }
         $address["country"] = $country;
 
-        $state = SimplUtil::state_name_from_code($address["state"]);
-        if(!isset($state)) {
-            throw new SimplCustomHttpBadRequest("state is not supported");
+        if(isset($address["state"]) && $address["state"] != "") {
+            $state = SimplUtil::state_name_from_code($address["state"]);
+            if(!isset($state)) {
```
Comment (sathia27) [inline includes/simpl_integration/class-cart-response.php:68]: won't this return empty? din't we need empty check here as well.

File: includes/simpl_integration/class-cart-response.php [hunk 2]
```diff
@@ -63,11 +63,13 @@ protected function convert_address_response($address) {
         }
         $address["country"] = $country;
 
-        $state = SimplUtil::state_name_from_code($address["state"]);
-        if(!isset($state)) {
-            throw new SimplCustomHttpBadRequest("state is not supported");
+        if(isset($address["state"]) && $address["state"] != "") {
```
Comment (sathia27) [inline includes/simpl_integration/class-cart-response.php:66]: instead of `isset` can we use `empty`. i guess `empty` checks both null and empty string

**Review by sathia27 — CHANGES_REQUESTED** [review]

File: includes/simpl_integration/class-cart-response.php (hunk 2 above)
Comment (rithiksharma100) [reply includes/simpl_integration/class-cart-response.php:66]: sure

Comment (rithiksharma100) [reply includes/simpl_integration/class-cart-response.php]: done

File: includes/simpl_integration/class-cart-response.php (hunk 1 above)
Comment (rithiksharma100) [reply includes/simpl_integration/class-cart-response.php]: added

**Review by sahil-simpl — APPROVED** [review]

**Review by sathia27 — APPROVED** [review]

**Pull request merged**